### DISCUSSION TITLE
Web front end for telnet client

### DISCUSSION THREAD

**Discussion by IBBBS** [NuSkooler/enigma-bbs#351] [General]
So I LOVE LOVE LOVE areas like Throwback BBS that allow TELNET on the webpage's front end. I saw that xibalba is looking at this too. I'm hopeful in seeing this come to fruition? It provides all that more exposure of BBS (represent!). Watching that closely (but no clue yet how to implement on my own).

**Comment by NuSkooler** [comment]
This already exists in a number of forms -- fTelnet, VTX, etc. Not something ENiGMA itself would provide. Throwback BBS is using fTelnet. ENiGMA supports these via the WebSocket server in which these clients connect to.

**Reply by NuSkooler** [reply]
[VTX client](https://github.com/codewar65/VTX_ClientServer/blob/master/www/vtxclient.js) (What https://xibalba.l33t.codes uses)
[fTelnet](https://www.ftelnet.ca/)

**Reply by IBBBS** [reply]
Yup - got it. Perfect!

**Reply by NuSkooler** [reply]
If you need help setting it up shout out in IRC or what not and we'll get you going

**Comment by IBBBS** [comment]
Nope - I'm good! It's running like a champ!
www.westwoodbbs.net

I am trying to figure out another thing - and if you're interested, I might
send you the code to see what the hell I'm doing wrong.

I'm trying to add "/ME" in mrc.js to process "<user> <action>". I have
tried several different code approaches and it just crashes mrc.js when
attempting it.

I saved both code approaches.
It's really just me not understanding node.js yet, but I'm getting there -
using other examples from similar code instances.

Enigma is rocking. I have four themes running smoothly for the most part. I
had one theme that would crash Enigma (no reason given - even with trace).
I still have it, but it's disabled at the moment. I think the only other
thing I struggle with (and it appears some others are too) is figuring out
how to get a different theme work in the message area (trying to properly
fit FTR/BODY/HDR all together without messing something up. Biggest problem
is seeing the BODY portion jump off to the right of the screen.

Other than that, it's running awesome.  I'm loving this product! Good stuff.
~ Nuke

On Sat, Feb 20, 2021 at 7:05 PM Bryan Ashby <notifications@github.com>
wrote:

> If you need help setting it up shout out in IRC or what not and we'll get
> you going
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NuSkooler/enigma-bbs/discussions/351#discussioncomment-389548>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASUPVMYZ2QKTAE2VUJBZ6IDTABE3NANCNFSM4XP2FPGA>
> .
>


-- 
*// WESTWOOD BBS II // SYSOP // telnet WestwoodBBS.net 8888*

**Reply by NuSkooler** [reply]
@WestwoodBBS I can't help a whole lot with "I tried something and it didn't work". For development of a module I suggest running a separate instance of enigma in a development environment and under a debugger (e.g. VSCode).

You'll notice the body art has a `%MT1` (the `MultiLineTextEditView` position) and a `%XY2` code in it -- this tells the system where the bottom of the editor area (the `%MT1`) is. The header + body + footer can't be more than the users term. It's generally best to stick within 79x24 to work well with all terminals. It's a bit finicky and something that will be re-written at some point, but the general rules would be the same.
